File: src/main/java/space/accident/api/interfaces/tileentity/IUpgradableMachine.java
package space.accident.api.interfaces.tileentity;

public interface IUpgradableMachine extends IMachineProgress {
	/**
	 * Accepts Upgrades. Some Machines have an Upgrade Limit.
	 */
	boolean isUpgradable();
	
	/**
	 * Accepts Muffler Upgrades
	 */
	boolean isMufflerUpgradable();
	
	/**
	 * Adds Muffler Upgrade
	 */
	boolean addMufflerUpgrade();
	
	/**
	 * Does this Machine have an Muffler
	 */
	boolean hasMufflerUpgrade();
}

File: src/main/kotlin/space/accident/api/interfaces/te/IUpgradableMachine.kt
package space.accident.api.interfaces.te

interface IUpgradableMachine : IMachineProgress {
    /**
     * Accepts Upgrades. Some Machines have an Upgrade Limit.
     */
    fun isUpgradable(): Boolean

    /**
     * Accepts Muffler Upgrades
     */
    fun isMufflerUpgradable(): Boolean

    /**
     * Accepts Steam-Converter Upgrades
     */
    fun isSteamEngineUpgradable(): Boolean

    /**
     * Adds Muffler Upgrade
     */
    fun addMufflerUpgrade(): Boolean

    /**
     * Adds MJ-Converter Upgrade
     */
    fun addSteamEngineUpgrade(): Boolean

    /**
     * Does this Machine have an Muffler
     */
    fun hasMufflerUpgrade(): Boolean

    /**
     * Does this Machine have a Steam-Converter
     */
    fun hasSteamEngineUpgrade(): Boolean
}